src/subscription.ts

```diff
@@ -1,125 +1,121 @@
 import { Vendor } from '.'
-import { ethers } from 'ethers'
+import { ethers, ContractTransaction } from 'ethers'
 import Deployed from './abstracts/deployed'
 import { PAYMENT_ABI, ERC20_ABI, SUBSCRIPTION_PAYMENT_ABI, SUBSCRIPTION_DATA_ABI } from './constants'
 import { INVALID_BICONOMY_KEY } from './errors'
 import { SubscriptionParameters, TokenData, TxResponse } from './interfaces'
 
-export default class extends Deployed {
+export default class SubscriptionContract extends Deployed {
   /**
+   * Creates an instance of SubscriptionContract.
    * @param vendor - Instance of a Vendor class
    */
   constructor(vendor: Vendor) {
     super(vendor, PAYMENT_ABI, ERC20_ABI, SUBSCRIPTION_PAYMENT_ABI, SUBSCRIPTION_DATA_ABI)
   }
 
   /**
-   * @remarks
-   * This method can be used to updated address of vault/escrow account
-   *
-   * @param a - address of escrow contract(vault)
+   * Updates the address of the vault/escrow account.
+   * @param escrowAddress - Address of the escrow contract (vault)
+   * @returns A promise that resolves to a transaction response.
    */
-  async updateEscrow(a: string): Promise<TxResponse> {
-    return await this.subscriptionDataContract?.functions.updateEscrow(a)
+  async updateEscrow(escrowAddress: string): Promise<TxResponse> {
+    return await this.paymentsContract?.functions.updateEscrow(escrowAddress)
   }
 
   /**
-   * @remarks
-   * This method can be used to updated address of staked token
-   *
-   * @param a - address of escrow contract(vault)
+   * Updates the address of the staked token.
+   * @param stakedTokenAddress - Address of the staked token.
+   * @returns A promise that resolves to a transaction response.
    */
-  async updateStakedToken(a: string): Promise<TxResponse> {
-    return await this.subscriptionDataContract?.functions.updateStakedToken(a)
+  async updateStakedToken(stakedTokenAddress: string): Promise<ContractTransaction> {
```

Comment:
why is there a different interface used here whereas everywhere its `TxResponse`?